provider/packet/userdata.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package packet
+
+import (
+	"github.com/juju/errors"
+	jujuos "github.com/juju/os/v2"
+
+	"github.com/juju/juju/cloudconfig/cloudinit"
+	"github.com/juju/juju/cloudconfig/providerinit/renderers"
+)
+
+type PacketRenderer struct{}
+
+func (PacketRenderer) Render(cfg cloudinit.CloudConfig, os jujuos.OSType) ([]byte, error) {
+	switch os {
+	case jujuos.Ubuntu:
```

Comment:
Interestingly, when I tried to bootstrap with the current HEAD of this PR, I saw the correct machine agent running but the lxd package was not installed. My theory here is that this is currently breaking due to the incorrect filtering of tags (juju was trying to connect to the wrong controller IP; it was targeting one of the model instances that are running inside the project instead of the controller that got created by my juju client)